fabric-lifecycle-events-v1/src/main/java/net/fabricmc/fabric/api/event/lifecycle/v1/ServerLifecycleEvents.java

```diff
@@ -0,0 +1,246 @@
+				callback.onChunkUnload(serverWorld, chunk);
+			}
+		}
+	});
+
+	/**
+	 * Called when an BlockEntity is loaded into a ServerWorld.
+	 */
+	public static final Event<BlockEntityLoadCallback<ServerWorld>> BLOCK_ENTITY_LOAD = EventFactory.createArrayBacked(BlockEntityLoadCallback.class, callbacks -> (blockEntity, world) -> {
+		if (EventFactory.isProfilingEnabled()) {
+			final Profiler profiler = world.getProfiler();
+			profiler.push("fabricServerBlockEntityLoad");
+
+			for (BlockEntityLoadCallback<ServerWorld> callback : callbacks) {
+				profiler.push(EventFactory.getHandlerName(callback));
+				callback.onLoadBlockEntity(blockEntity, world);
+				profiler.pop();
+			}
+
+			profiler.pop();
+		} else {
+			for (BlockEntityLoadCallback<ServerWorld> callback : callbacks) {
+				callback.onLoadBlockEntity(blockEntity, world);
+			}
+		}
+	});
+
+	/**
+	 * Called when an BlockEntity is unloaded from a ServerWorld.
+	 */
+	public static final Event<BlockEntityUnloadCallback<ServerWorld>> BLOCK_ENTITY_UNLOAD = EventFactory.createArrayBacked(BlockEntityUnloadCallback.class, callbacks -> (blockEntity, world) -> {
+		if (EventFactory.isProfilingEnabled()) {
+			final Profiler profiler = world.getProfiler();
+			profiler.push("fabricServerBlockEntityUnload");
+
+			for (BlockEntityUnloadCallback<ServerWorld> callback : callbacks) {
+				profiler.push(EventFactory.getHandlerName(callback));
+				callback.onUnloadBlockEntity(blockEntity, world);
+				profiler.pop();
+			}
+
+			profiler.pop();
+		} else {
+			for (BlockEntityUnloadCallback<ServerWorld> callback : callbacks) {
+				callback.onUnloadBlockEntity(blockEntity, world);
+			}
+		}
+	});
+
+	/**
+	 * Called when an Entity is loaded into a ServerWorld.
+	 *
+	 * <p>Note there is no corresponding unload event because entity unloads cannot be reliably tracked.
+	 */
+	public static final Event<EntityLoadCallback<ServerWorld>> ENTITY_LOAD = EventFactory.createArrayBacked(EntityLoadCallback.class, callbacks -> (entity, world) -> {
+		if (EventFactory.isProfilingEnabled()) {
+			final Profiler profiler = world.getProfiler();
+			profiler.push("fabricServerEntityLoad");
+
+			for (EntityLoadCallback<ServerWorld> callback : callbacks) {
+				profiler.push(EventFactory.getHandlerName(callback));
+				callback.onEntityLoad(entity, world);
+				profiler.pop();
+			}
+
+			profiler.pop();
+		} else {
+			for (EntityLoadCallback<ServerWorld> callback : callbacks) {
+				callback.onEntityLoad(entity, world);
+			}
+		}
+	});
+
+	/**
+	 * Called when the server has started. At this stage, all worlds are live.
+	 */
+	public static final Event<GameLifecycleCallback<MinecraftServer>> SERVER_STARTED = EventFactory.createArrayBacked(GameLifecycleCallback.class, (callbacks) -> (server) -> {
+		for (GameLifecycleCallback<MinecraftServer> callback : callbacks) {
+			callback.onChangeLifecycle(server);
+		}
+	});
+
+	/**
+	 * Called when the server has started stopping. All worlds are still present.
```

Comment:
This could clarify what stopping means, especially with the client having a less permanent stop (in a way).